src/Core/Managed/Shared/Extensibility/Implementation/SimpleMetricStatisticsAggregator.cs

```diff
@@ -0,0 +1,116 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.Implementation
+{
+    using System;
+    using System.Threading;
+
+    /// <summary>
+    /// Represents mechanism to calculate basic statistical parameters of a series of numeric values.
+    /// </summary>
+    internal class SimpleMetricStatisticsAggregator
+    {
```

Comment:
Yes, given the read pattern (once right after snapshot) I do not think this is a problem. I do not want to complicate getters with honoring the lock especially considering the fact that they'll be called sequentially.